be/src/runtime/int128_arithmetics_x86_64.h

```diff
@@ -93,15 +93,40 @@ static int asm_mul(T x, U y, Int128Wrapper& res) {
     } else {
         __asm__ __volatile__(
                 "mov %[x], %%rax\n\t"
-                "imul %[y], %%rax\n\t"
+                "imul %[y]\n\t"
+                "mov %%rdx, %[high]\n\t"
                 "mov %%rax, %[low]"
-                : [ low ] "=r"(res.s.low), "=@cco"(overflow)
+                : [ high ] "=r"(res.s.high), [ low ] "=r"(res.s.low), "=@cco"(overflow)
                 : [ x ] "r"(x), [ y ] "r"(y)
                 : "cc", "rdx", "rax");
     }
     return overflow;
 }
 
+static int64_t asm_mul32(int32_t x, int32_t y) {
```

Comment:
Is it works on ARM ?

Reply:
asm_mul32 is enclosed by X86_64 macro finally. this case is not optimized on arm.